### ISSUE TITLE
bug: results.fileNames undefined during compilation in stencil 2.17.3 in NX

### ISSUE THREAD

**Issue by orl99** [ionic-team/stencil#3510]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.17.3

### Current Behavior

Currently, I we are having problems compiling one library that we recently created with a few components for a future DS that we plan to make, this is currently the CLI out when we run ` nx run aqua-elements:build`

<img width="1037" alt="image" src="https://user-images.githubusercontent.com/13844071/182925188-af1b3d46-4b48-4bc7-8719-3cc92e805df7.png">

```sh

nx run aqua-elements:build


[ WARN  ]  exit 1

/Users/qismbp12/Documents/GitHub/cbhz-nx/node_modules/@stencil/core/compiler/stencil.js:41837
    results.fileNames = results.fileNames.filter((f) => {
                                ^

TypeError: Cannot read properties of undefined (reading 'fileNames')
    at Object.t.getParsedCommandLineOfConfigFile (/Users/qismbp12/Documents/GitHub/cbhz-nx/node_modules/@stencil/core/compiler/stencil.js:41837:33)
    at pe (/Users/qismbp12/Documents/GitHub/cbhz-nx/node_modules/@stencil/core/compiler/stencil.js:8913:2294592)
    at Object.e.createWatchProgram (/Users/qismbp12/Documents/GitHub/cbhz-nx/node_modules/@stencil/core/compiler/stencil.js:8913:2287601)
    at createTsBuildProgram (/Users/qismbp12/Documents/GitHub/cbhz-nx/node_modules/@stencil/core/compiler/stencil.js:63449:12)
    at /Users/qismbp12/Documents/GitHub/cbhz-nx/node_modules/@stencil/core/compiler/stencil.js:63483:5
    at new Promise (<anonymous>)
    at createFullBuild (/Users/qismbp12/Documents/GitHub/cbhz-nx/node_modules/@stencil/core/compiler/stencil.js:63459:10)
    at Object.build (/Users/qismbp12/Documents/GitHub/cbhz-nx/node_modules/@stencil/core/compiler/stencil.js:64732:23)
    at taskBuild (/Users/qismbp12/Documents/GitHub/cbhz-nx/node_modules/@stencil/core/cli/index.cjs:1671:40)
    at async runTask (/Users/qismbp12/Documents/GitHub/cbhz-nx/node_modules/@stencil/core/cli/index.cjs:2327:13)
```
<img width="1037" alt="image" src="https://user-images.githubusercontent.com/13844071/182928026-def95536-7576-4af2-bc27-80a96686b656.png">

It's important to mention that we use NX as monorepo.

Any reasons why we are having this problem?
Is anyone also having the same problem?

### Expected Behavior

Have a successful compilation process 

### Steps to Reproduce

1- `npm i `
2- `nx run aqua-elements:build`

### Code Reproduction URL

N/A

**Comment by rwaskiewicz** [comment]
Hey @orl99 👋 

The value stored in `results` there is determined by a call to a TypeScript compiler API, `getParsedCommandLineOfConfigFiles`. That code can be found [here](https://github.com/microsoft/TypeScript/blob/b8f648832379005fc8c3c9b34cc5e4acd01653e4/src/compiler/commandLineParser.ts#L1864-L1894). Looking around that source, it appears as it [this block](https://github.com/microsoft/TypeScript/blob/b8f648832379005fc8c3c9b34cc5e4acd01653e4/src/compiler/commandLineParser.ts#L1872-L1876) might be the offending code that's returning `undefined`. 

I _think_ this could be related to attempting to resolve a TS configuration file, which may make sense if the file is in a monorepo. 

I think that Stencil can handle this a little more gracefully, but in order to continue debugging this further, can you create a minimal reproduction case for us to investigate further?

**Comment by m4s7** [comment]
Thanks for this thread, I have the same problem. I just want to link the related issue in the @nxext/stencil repo: https://github.com/nxext/nx-extensions/issues/771

**Comment by m4s7** [comment]
Steps to reproduce:

```console
yarn create nx-workspace
```
- Name: test-stencil
- Type: app
- nx-cloud: no

```console
yarn add --dev @nxext/stencil
````

```console
yarn nx generate @nxext/stencil:application sample --no-interactive
```

```console
yarn nx run sample:build
```

-> Working build

```console
yarn add --dev @stencil/core@2.17.2
```

```console
yarn nx run sample:build
```
-> Failing Build


Repo: https://github.com/m4s7/failing-sample

**Comment by pranav-js** [comment]
only way to fix as of now is to pin stencil/core at 2.17.1 , as 2.17.2 onwards it starts giving this error

**Comment by rwaskiewicz** [comment]
Thanks all, y'all are awesome! I've confirmed this is a bug, and relates to how the NX plugin calls Stencil's APIs. I've ID'ed the commit in the Stencil repo that causes this, and am going to have someone take a look at this early next week

**Comment by rwaskiewicz** [comment]
The PR that fixes this issue (#3518) is a part of [today's v2.17.4 release](https://github.com/ionic-team/stencil/releases/tag/v2.17.4). As a result, I'm going to close this issue. Should the issue reoccur on v2.17.4+ of Stencil, please open a new ticket, thanks!
